authors/apps/authentication/views.py

```diff
@@ -73,3 +89,74 @@ def update(self, request, *args, **kwargs):
 
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+@api_view(['GET', 'POST'])
+def password_reset(request):
+    if request.method == 'POST':
+ 
+        token = uuid.uuid4()
+        email = request.data.get('email')
+
+        user = get_object_or_404(User, email=email)
+        
+        Token.objects.create(
+            token=token,
+            email=user.email
+        )
+
+        url = f'http://127.0.0.1:8000/api/reset_password/?token={token}'
+        
+        message = f'Please click the link to reset your password\n {url} \n {token}'
+
+        send_mail(
+            'Password Reset Link',
+            message,
+            settings.EMAIL_HOST_USER,
+            [user.email],
+            fail_silently=False,
+        )
+
+        return Response({"message": "email sent", "data": request.data})
+    return Response({"message": "Great Work, TeamOdin"})
+
+@api_view(['GET'])
+def reset_password(request):
```

Comment:
For consistency, I would prefer we stick to using the class-based API views for example RegistrationAPIView instead of the decorators